### PR TITLE
OCM-11026 | test: automate id 38775

### PR DESCRIPTION
**Pull request by OAharoni-RedHat**
```
oaharoni-mac:rosa oaharoni$ ginkgo run -focus 38775 tests/e2e
Running Suite: ROSA CLI e2e tests suite - /Users/oaharoni/rosa_workspace/rosa/tests/e2e
=======================================================================================
Random Seed: 1726001029

Will run 1 of 220 specs
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS•SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS

Ran 1 of 220 Specs in 30.703 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 219 Skipped
PASS

Ginkgo ran 1 suite in 36.769029833s
Test Suite Passed
```

### CODE REVIEW

File: tests/utils/exec/rosacli/machinepool_service.go [hunk 1]
```diff
@@ -162,6 +163,14 @@ func (m *machinepoolService) CreateMachinePool(
 	return
 }
 
+// Create MachinePool with no flags. Only used for validation in id:38775
+func (m *machinepoolService) CreateMachinePoolEmpty() (output bytes.Buffer, err error) {
```
Comment (OAharoni-RedHat) [inline tests/utils/exec/rosacli/machinepool_service.go:167]: Not sure if this was the best practice for writing this part of the test case, please let me know if you have other ideas/suggestions.

Comment (xueli181114) [reply tests/utils/exec/rosacli/machinepool_service.go:167]: Let's not define such a function like this. Better to enhance function `CreateMachinePool` to make it support missing required flag.

File: tests/e2e/test_rosacli_machine_pool.go [hunk 2]
```diff
@@ -767,6 +767,85 @@ var _ = Describe("Create machinepool",
...
+				labels.Runtime.Day2, labels.Medium,
+				func() {
+					mpName := "mp-38775"
+					fakeClusterName := "fake_38775"
+					By("Create machine pool with no flags")
+					output, err := rosaClient.MachinePool.CreateMachinePoolEmpty()
+					Expect(err).To(HaveOccurred())
+					Expect(output.String()).Should(ContainSubstring("required flag(s) \"cluster\" not set"))
+
+					By("Create machine pool with invalid values")
```
Comment (xueli181114) [inline tests/e2e/test_rosacli_machine_pool.go:780]: Better to indicate the validation checkpoint in `By`
```suggestion
					By("Create machine pool to non-exist cluster")
```

File: tests/e2e/test_rosacli_machine_pool.go [hunk 3]
```diff
@@ -767,6 +767,85 @@ var _ = Describe("Create machinepool",
...
+					output, err = rosaClient.MachinePool.CreateMachinePool(fakeClusterName, mpName)
+					Expect(err).To(HaveOccurred())
+					Expect(output.String()).Should(
+						ContainSubstring("There is no cluster with identifier or name '%s'", fakeClusterName))
+
+					output, err = rosaClient.MachinePool.CreateMachinePool(
```
Comment (xueli181114) [inline tests/e2e/test_rosacli_machine_pool.go:786]: Add `By` to indicate the step we are running. It is convenient for future's debug

**Review by radtriste — APPROVED** [review]
/lgtm

**Pull request merged**